src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +76,35 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (has(POD_DELETION_TIMESTAMP)) {
```

Comment:
not complaining about your code here: to me all this feels like the backend is inconsistent and should get fixed. I believe that the (server reported) json is missing an (additional) unique field where the pod state is reflected. The whole method inspects different locations and merge these to exactly this: a single and unique field for the pod state. Shouldn't we file an issue to openshift/origin?